src/libstd/sync/mutex.rs

```diff
@@ -107,6 +107,69 @@ use crate::sys_common::poison::{self, LockResult, TryLockError, TryLockResult};
 ///
 /// *guard += 1;
 /// ```
+///
+/// It is sometimes a good idea (or even necessary) to manually drop the mutex
+/// to unlock it as soon as possible. If you need the resource until the end of
+/// the scope, this is not needed.
+///
+/// ```
+/// use std::sync::{Arc, Mutex};
```

Comment:
I like this example, but I also feel like it is somewhat long-winded. Using a dead-lock with thread-joining is a great way to demonstrate the problem, and the amount of commentary in other parts of the code (especially within the thread, detracts from the point slightly.

Reply:
I can remove most of the commentary, no problem. Since it was intended to be a warning about deadlocks, I made a lot of commentary to really drive the point home, but I can certainly remove/condense the comments.